### PR TITLE
Add ASDK prechecker script

### PR DESCRIPTION
**Pull request by PatAltimore**
Add ASDK prechecker script to deployment folder.

### CODE REVIEW

File: Deployment/asdk-prechecker.ps1 [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+        write-host -ForegroundColor red "["(date -format "HH:mm:ss")"]" " -- Failed to connect to AAD endpoint 'https://login.windows.net'."
+        $Global:ChecksFailure++
+    }
+}
+
+function CheckSystemDisk {
+    write-host -ForegroundColor yellow "["(date -format "HH:mm:ss")"]" "Checking system disk capacity..."
+
+    $systemDisk = Get-Disk | ? {$_.IsSystem -eq $true}
+    If ($systemDisk.Size -lt 180 * 1024 * 1024 * 1024)
```
Comment (peiwentian) [inline Deployment/asdk-prechecker.ps1:75]: Currently the system requires 200GB system disk, please update to
`200 * 1024 * 1024 * 1024`

https://docs.microsoft.com/en-us/azure-stack/asdk/asdk-deploy-considerations?view=azs-2008

File: Deployment/asdk-prechecker.ps1 [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " Checking free space for extracting the ASDK files..."
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Listing disks and their free space"
+    write-host -ForegroundColor gray (Get-Disk | Get-Partition | Get-Volume | Sort-Object -Property SizeRemaining -Descending | Out-String)
+    $volumes = (Get-disk | ? {$_.BusType -ne 'File Backed Virtual' -or $_.IsBoot} | Get-Partition | Get-Volume |`
+         ? {-not [String]::IsNullOrEmpty($_.DriveLetter)} | sort -Property SizeRemaining -Descending)
+    if (!$volumes -or ($volumes | Measure-Object).count -le 0) {
+        Write-Host -ForegroundColor red "["(date -format "HH:mm:ss")"]" " -- Free space check failed. No volumes are available."
+        $Global:ChecksFailure++
+    }
+    if ($volumes[0].SizeRemaining -lt 120 * 1024 * 1024 * 1024) {
```
Comment (peiwentian) [inline Deployment/asdk-prechecker.ps1:120]: Current CloudBuilder.vhdx will be 130GB. Please update to 
`if ($volumes[0].SizeRemaining -lt 130 * 1024 * 1024 * 1024) {`

File: Deployment/asdk-prechecker.ps1 [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+    }
+}
+
+function CheckRam {
+    write-host -ForegroundColor yellow "["(date -format "HH:mm:ss")"]" "Checking Memory..."
+    
+    $mem = Get-WmiObject -Class Win32_ComputerSystem
+    $totalMemoryInGB = [Math]::Round($mem.TotalPhysicalMemory / (1024 * 1024 * 1024))
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Memory on this server = $totalMemoryInGB"
+    if ($totalMemoryInGB -lt 96) {
```
Comment (peiwentian) [inline Deployment/asdk-prechecker.ps1:137]: Currently the memory for ASDK has 192-GB minimum and 256-GB recommended.
Please update the value accordingly.

File: Deployment/asdk-prechecker.ps1 [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+function CheckCPU {
+
+    write-host -ForegroundColor yellow "["(date -format "HH:mm:ss")"]" "Checking processor information..."
+
+    $CPUCount = (Get-WmiObject -class win32_processor –computername localhost).count
+    $CoreCount =  ((Get-WmiObject -class win32_processor –computername localhost -Property "numberOfCores")[0].numberOfCores)*$CPUCount
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Number of CPU sockets = $CPUCount"
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Number of physical cores =  $CoreCount"
+
+    If (($CPUCount -lt 2) -or ($CoreCount -lt 12)){
```
Comment (peiwentian) [inline Deployment/asdk-prechecker.ps1:254]: Currently the CPU requirement for ASDK is with "16" minimum and "20" recommended. Please update accordingly

File: Deployment/asdk-prechecker.ps1 (hunk 1 above)
Comment (PatAltimore) [reply Deployment/asdk-prechecker.ps1]: Updated system calculation and error message. Added reference to documentation in the description section at the top of the script.

File: Deployment/asdk-prechecker.ps1 (hunk 2 above)
Comment (PatAltimore) [reply Deployment/asdk-prechecker.ps1]: Updated calculation.

File: Deployment/asdk-prechecker.ps1 (hunk 3 above)
Comment (PatAltimore) [reply Deployment/asdk-prechecker.ps1]: Changed test condition to `$totalMemoryInGB -lt 192`.  Improved error message description.

File: Deployment/asdk-prechecker.ps1 (hunk 4 above)
Comment (PatAltimore) [reply Deployment/asdk-prechecker.ps1]: Changed test condition to `$CoreCount -lt 16`. Updated error message description.

**Review by PatAltimore — COMMENTED** [review]
Thanks @peiwentian. I've made the recommended changes.

**Comment by peiwentian** [comment]
@PatAltimore I have updated the script, it looks good to me.

**Pull request merged**